File: Set 7/Sprial Matrix/Matrix.py
def sprial_matrix(matrix):
    top, bottom = 0, len(matrix) - 1
    left, right = 0, len(matrix[0]) - 1

    while top <= bottom and left <= right:
        for i in range(left, right + 1):
            print(matrix[top][i], end=" ")
        top += 1

        for i in range(top, bottom + 1):
            print(matrix[i][right], end=" ")
        right -= 1

        if top <= bottom:
            for i in range(right, left - 1, -1):
                print(matrix[bottom][i], end=" ")
            bottom -= 1

        if left <= right:
            for i in range(bottom, top - 1, -1):
                print(matrix[i][left], end=" ")
            left += 1
    print()


t = int(input().strip())
for _ in range(t):
    matrix = []
    while True:
        line = input().strip()
        if not line:
            break
        matrix.append(list(map(int, line.split())))
    sprial_matrix(matrix)

File: Set 7/Sprial Matrix/Matrix.cpp
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;

void Print(const vector<vector<int>>& matrix) {
    int top = 0, bottom = matrix.size() - 1;
    int left = 0, right = matrix[0].size() - 1;

    while (top <= bottom && left <= right) {
        for (int i = left; i <= right; i++) {
            cout << matrix[top][i] << " ";
        }
        top++;

        for (int i = top; i <= bottom; i++) {
            cout << matrix[i][right] << " ";
        }
        right--;

        if (top <= bottom) {
            for (int i = right; i >= left; i--) {
                cout << matrix[bottom][i] << " ";
            }
            bottom--;
        }

        if (left <= right) {
            for (int i = bottom; i >= top; i--) {
                cout << matrix[i][left] << " ";
            }
            left++;
        }
    }
    cout << endl;
}

int main() {
    int t;
    cin >> t;
    cin.ignore(); 

    while (t-- > 0) {
        string matrixInput;
        string line;
        while (getline(cin, line)) {
            if (line.empty()) break;
            matrixInput += line + "\n";
        }

        stringstream ss(matrixInput);
        string row;
        vector<vector<int>> matrix;

        while (getline(ss, row)) {
            stringstream rowStream(row);
            vector<int> values;
            int value;
            while (rowStream >> value) {
                values.push_back(value);
            }
            matrix.push_back(values);
        }

        Print(matrix);
    }

    return 0;
}
